modules/custom/falcon_thankq/src/Plugin/Field/FieldWidget/InlineSourceCodeForm.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Drupal\falcon_thankq\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element;
+use Drupal\inline_entity_form\Plugin\Field\FieldWidget\InlineEntityFormComplex;
+
+/**
+ * Source codes field widget.
+ *
+ * @FieldWidget(
+ *   id = "inline_source_code_form",
+ *   label = @Translation("Inline Source Codes form"),
+ *   field_types = {
+ *     "entity_reference"
+ *   },
+ *   multiple_values = true
+ * )
+ */
+class InlineSourceCodeForm extends InlineEntityFormComplex {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+
+    // Mainly use parent code to render the form.
+    $element = parent::formElement($items, $delta, $element, $form, $form_state);
+
+    // Change field label to the one related to source codes.
+    $element['entities']['#table_fields']['label']['label'] = t('Source code');
+
+    foreach (Element::children($element['entities']) as $key) {
+
+      // Get Source Code entity.
+      /* @var $entity \Drupal\eck\EckEntityInterface */
+      $entity = $element['entities'][$key]['#entity'];
+
+      // Grab source code & human readable label.
+      $source_code = $entity->get('field_source_code')->getString();
+      $label = $entity->get('field_label')->getString();
+      $date = '';
+
+      // Render availability date in the human readable format.
+      $availability = $entity->get('field_availability')->getValue();
+      if (!empty($availability[0]['value'])) {
+        $date .= format_string('from @from ', ['@from' => $availability[0]['value']]);
```

Comment:
format_string() is deprecated - https://api.drupal.org/api/drupal/core%21includes%21bootstrap.inc/function/format_string/8.5.x - looks like we need to use https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Render%21FormattableMarkup.php/function/FormattableMarkup%3A%3AplaceholderFormat/8.5.x

Reply:
@duozersk agree, fixed.